### ISSUE TITLE
Update Documentation

### ISSUE THREAD

**Issue by fmasion** [japgolly/scalajs-react#271]
The exemple code in the doc doesn't seam to work : 
https://github.com/japgolly/scalajs-react/blob/master/doc/USAGE.md#creating-components

```
val Hello =
  ReactComponentB[String]("Hello <name>")
    .render(name => <.div("Hello ", name))
    .build
// Usage:
Hello("Draconus")
```

will fail to compile

It will work with render_P instead

**Comment by PerWiklander** [comment]
Maybe this:

```
val Hello =
  ReactComponentB[String]("Hello <name>")
    .render($ => <.div("Hello ", $.props))
    .build
// Usage:
Hello("Draconus")
```

`$` is a fairly standard name (in this library) for the `componentScope`. If you come from a JS background `$.props` would be comparable to `this.props`.

Not having ` <name>` in the component name and then using it as variable name in render might lessen confusion for first timers as well.

**Comment by fmasion** [comment]
I agree with your comment : <name> is confusing (at least with this syntax) as it looks like some kind of jsx

**Comment by japgolly** [comment]
Cheers, those are both good points. I'll just quickly fix that now...

**Comment by japgolly** [comment]
Closed via ecc11f7. Please have another look and let me know or PR if there's anything else. Thanks.
